src/nvim/eval.c

```diff
@@ -11686,14 +11691,70 @@ static void f_jobstart(typval_T *argvars, typval_T *rettv, FunPtr fptr)
         return;
       }
     }
+    dictitem_T *item;
+    // or new_env
+    item = tv_dict_find(job_opts, S_LEN("env"));
+    if (item) {
+      size_t env_size = 0; // len ?
+      size_t i = 0;
+      // size_t orig_env_size = 0;
+      if (item->di_tv.v_type != VAR_DICT) {
+        EMSG2(_(e_invarg2), "Expected a dictionary for 'env'");
+        return;
+      }
+
+      // number of items in dict
+      env_size += (size_t)tv_dict_len(item->di_tv.vval.v_dict);
+      ILOG("%d passed env variables", env_size);
+
+      if (!reset_env) {
+        char **environ = os_getfullenv();
```

Comment:
Now I get it. Here we actually take action to attempt to _gather_ the full environment and pass it on. That's rather different than what is implied: "reset_env" sounds like it takes action to _delete_ an environment that would otherwise be inherited.

Something like `include_parent_env` would be better, but too long ... so I'll be quiet.